src/crypto/a128kw.rs

```diff
@@ -79,3 +118,18 @@ impl JweA128KWEncipher {
         })
     }
 }
+
+fn kid(wrap_key: &Aes128Key) -> String {
+    let mut skey = HmacSha256Key::default();
+    let skey_slice = skey.as_mut_slice();
+    let wrap_key_slice = wrap_key.as_slice();
+    let skey_slice_mut = &mut skey_slice[..wrap_key_slice.len()];
+    skey_slice_mut.copy_from_slice(wrap_key_slice);
```

Comment:
... we're taking a wrap-key-length from skey_slice, then adding wrap-key on the end of it? seems weird?

Reply:
It's the only way to copy the key into the slice, you can't view the key bytes any other way.